src/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/AuthenticationHelper.cs

```diff
@@ -251,11 +252,19 @@ await TrySetDigestAuthToken(request, challenge.Credential, digestResponse, isPro
                                     {
                                         try
                                         {
+                                            if (NetEventSource.IsEnabled)
+                                            {
+                                                NetEventSource.Info(pool.PreAuthCredentials, $"Adding Basic credential to cache, uri={authUri}, username={challenge.Credential.UserName}");
+                                            }
                                             pool.PreAuthCredentials.Add(authUri, BasicScheme, challenge.Credential);
                                         }
                                         catch (ArgumentException)
                                         {
                                             // The credential already existed.
+                                            if (NetEventSource.IsEnabled)
+                                            {
+                                                NetEventSource.Info(pool.PreAuthCredentials, $"Basic credential present in cache, uri={authUri}, username={challenge.Credential.UserName}");
+                                            }
```

Comment:
What additional information does logging this tell me?  I'm wondering if there was a TryAdd method if we'd be logging it returning false.  If yes, then fine.

Reply:
Debugging authentication problems tends to be hard especially when we use internal caches for the credentials. This seems like a useful place to add some logging to help us determine if our credential caching logic is working as intended.